drivers/src/mailbox.rs

```diff
@@ -331,10 +353,6 @@ impl MailboxRecvTxn {
         let mbox = mbox::RegisterBlock::mbox_csr();
         mbox.status().write(|w| w.status(|_| status));
 
-        // Release the lock
-        let mbox = mbox::RegisterBlock::mbox_csr();
-        mbox.execute().write(|w| w.execute(false));
-
         self.state = MailboxOpState::Idle;
```

Comment:
We are no longer transitioning to idle. We need to remove that line.

Reply:
How so? The write to the status() register transfer control back to the SoC. We're definitely not in an Execute state any longer.

Reply:
FWIW, the FSM state in the RTL transitions from EXECUTE_UC to EXECUTE_SOC.

FWIW, I'm considering getting rid of self.state and instead using the type system to maintain state. This will result in less code (no runtime state checks) and will be less error prone (invalid state transitions won't compile).

Reply:
The write to the execute register unlocks the mailbox. Did that change?

Reply:
statemachine! {
    transitions: {
        // CurrentState Event [guard] / action = NextState
        *Idle + RdLock(Owner) [is_not_locked] / lock = RdyForCmd,
        RdyForCmd  + CmdWrite(Cmd) / set_cmd = RdyForDlen,
        RdyForDlen + DlenWrite(DataLength) / init_dlen = RdyForData,
        RdyForData + DataWrite(DataIn) / enqueue = RdyForData,
        RdyForData + ExecSet = Exec,
        Exec + DataRead / dequeue = Exec,
        Exec + DlenWrite(DataLength) / init_dlen = Exec,
        Exec + DataWrite(DataIn) / enqueue = Exec,
        Exec + ExecClear [is_locked] / unlock = Idle
    }
}

Reply:
Yes, but that's done by the *sender*, not the *receiver*. From the perspective of the microcontroller, the transaction is over.

Reply:
The RTL makes a distinction between what the sender can do vs the receiver. The emulator peripheral does not make this distinction, and thus it allows things to happen that should not be allowed (I plan to fix this after I have robust tests against the RTL).